src/main/java/io/confluent/connect/jdbc/dialect/SqlServerDatabaseDialect.java

```diff
@@ -108,6 +108,25 @@ public SqlServerDatabaseDialect(AbstractConfig config) {
     jtdsDriver = jdbcUrlInfo == null ? false : jdbcUrlInfo.subprotocol().matches("jtds");
   }
 
+  @Override
+  public TableId parseTableIdentifier(String fqn) {
+    List<String> parts = identifierRules().parseQualifiedIdentifier(fqn);
+    if (parts.isEmpty()) {
+      throw new IllegalArgumentException("Invalid fully qualified name: '" + fqn + "'");
+    }
+    if (parts.size() == 1) {
+      return new TableId(null, "dbo", parts.get(0));
```

Comment:
WDYT about calling `super.parseTableIdentifier(fqn)` and then if the schema is null, returning a replacement TableId?